Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -320,9 +349,15 @@ public class PythonCompiler : SinglePassCompiler {
       var cw = (ClassWriter)CreateClass(IdProtect(nt.EnclosingModuleDefinition.CompileName), IdName(nt), nt, wr);
       var w = cw.MethodWriter;
       var udt = UserDefinedType.FromTopLevelDecl(nt.tok, nt);
-      var d = TypeInitializationValue(udt, wr, nt.tok, false, false);
-
-      w.WriteLine($"default = staticmethod(lambda: {d})");
+      if (nt.WitnessKind == SubsetTypeDecl.WKind.Compiled) {
+        w.WriteLine("@staticmethod");
+        var block = w.NewBlockPy("def default():");
+        var wStmts = block.Fork();
+        block.Write("return ");
+        TrExpr(nt.Witness, block, false, wStmts);
+      } else {
+        w.WriteLine($"default = staticmethod(lambda: {TypeInitializationValue(udt, wr, nt.tok, false, false)})");
+      }
```

Comment:
Can this be in a shared `DefineDefaultMethod` method?

Reply:
What would you like to be shared? The first one needs the block so we have something for `wStmts` to point to. The other way is more succinct.